commands/utils/packagehandlers/packagehandlers_test.go

```diff
@@ -1,27 +1,151 @@
+func TestMavenPackageHandler_UpdateImpactedPackage(t *testing.T) {
+	testDataDir := getTestDataDir(t)
+	mvn := MavenPackageHandler{
+		mavenDepToPropertyMap: map[string][]string{
+			"junit": {"junit:junit", "3.8.1"},
+		},
+	}
+	test := indirectPackageFixTest{
+		impactedPackage: "junit",
+		fixVersionInfo: &utils.FixVersionInfo{
+			FixVersion:       "4.11",
+			PackageType:      "maven",
+			DirectDependency: false,
+		},
+		shouldFix: false,
+	}
+	cleanup := createTempFolderAndChDir(t, testDataDir, coreutils.Maven)
+	defer cleanup()
+	shouldFix, err := mvn.UpdateImpactedPackage(test.impactedPackage, test.fixVersionInfo)
+	assert.NoError(t, err)
+	assert.Equal(t, test.shouldFix, shouldFix)
+}
+
+// Until implemented, test we are not attempting to fix indirect dependencies in python
+func TestPythonPackageHandler_UpdateImpactedPackage(t *testing.T) {
+	testDataDir := getTestDataDir(t)
+	testcases := []pythonIndirectPackages{
+		{
+			indirectPackageFixTest: indirectPackageFixTest{impactedPackage: "urllib3",
+				fixVersionInfo: &utils.FixVersionInfo{
+					FixVersion:       "1.25.9",
+					PackageType:      "pip",
+					DirectDependency: false}, shouldFix: false},
+			requirementsPath:     "requirements.txt",
+			pythonPackageManager: "pip",
+		}, {
+			indirectPackageFixTest: indirectPackageFixTest{impactedPackage: "urllib3",
+				fixVersionInfo: &utils.FixVersionInfo{
+					FixVersion:       "1.25.9",
+					PackageType:      "poetry",
+					DirectDependency: false}, shouldFix: false},
+			requirementsPath:     "pyproejct.toml",
+			pythonPackageManager: "poetry",
+		}, {
+			indirectPackageFixTest: indirectPackageFixTest{impactedPackage: "urllib3",
+				fixVersionInfo: &utils.FixVersionInfo{
+					FixVersion:       "1.25.9",
+					PackageType:      "pipenv",
+					DirectDependency: false}, shouldFix: false},
+			requirementsPath:     "Pipfile",
+			pythonPackageManager: "pipenv",
+		},
+	}
+	for _, test := range testcases {
+		t.Run(test.pythonPackageManager.ToString(), func(t *testing.T) {
+			packageHandler := GetCompatiblePackageHandler(test.fixVersionInfo, &utils.ScanDetails{
+				Project: utils.Project{PipRequirementsFile: test.requirementsPath}}, nil)
+			cleanup := createTempFolderAndChDir(t, testDataDir, test.fixVersionInfo.PackageType)
+			defer cleanup()
+			shouldFix, _ := packageHandler.UpdateImpactedPackage(test.impactedPackage, test.fixVersionInfo)
+			assert.Equal(t, test.shouldFix, shouldFix)
+		})
+	}
+}
+
+func getTestDataDir(t *testing.T) string {
+	testdataDir, err := filepath.Abs(filepath.Join("..", "..", "testdata/indirect-projects"))
+	assert.NoError(t, err)
+	return testdataDir
+}
+
+func createTempFolderAndChDir(t *testing.T, testdataDir string, tech coreutils.Technology) func() {
+	// Create temp technology project
+	projectPath := filepath.Join(testdataDir, tech.ToString())
+	tmpProjectPath, cleanup := testdatautils.CreateTestProject(t, projectPath)
+	assert.NoError(t, os.Chdir(tmpProjectPath))
```

Comment:
The cleanup function returns by this function should also change directory back to the current directory.

Reply:
As i talked to @sverdlov93 this isn't the expected behavior in `build-info-go` and could cause unexpected change dirs, that's why i added change dir at the end of each test.